CDP4Common/Poco/Thing.cs

```diff
@@ -339,6 +339,39 @@ public IEnumerable<Thing> QueryContainedThingsDeep()
             return temp;
         }
 
+        /// <summary>
+        /// Queries the referenced <see cref="Thing"/>s of the current <see cref="Thing"/>
+        /// </summary>
+        /// <remarks>
+        /// this does not include the contained <see cref="Thing"/>s, the contained <see cref="Thing"/>s
+        /// are exposed via the <see cref="ContainerLists"/> method
+        /// </remarks>
+        /// <returns></returns>
+        public virtual IEnumerable<Thing> QueryReferencedThings()
+        {
+            return Enumerable.Empty<Thing>();
+        }
+
+        /// <summary>
+        /// Queries all the referenced <see cref="Thing"/>s of the current <see cref="Thing"/>, along the complete 
+        /// referenced <see cref="Thing"/> chain and returns them as a flat list
+        /// </summary>
+        /// <returns>
+        /// An <see cref="IEnumerable{Thing}"/> 
+        /// </returns>
+        public IEnumerable<Thing> QueryReferencedThingsDeep()
+        {
+            var temp = new List<Thing>();
+
+            foreach (var thing in this.QueryReferencedThings())
```

Comment:
For `QueryReferencedThingsDeep`, I was thinking of returning the list of objects referenced from the full containment [sub]tree, not a tree of referenced things: so iterating through `this.ContainerList`, not the referenced things.

For the full tree of things "connected" to the current thing, we would indeed interate through both contained and referenced things.